tests/models/whisper/test_modeling_tf_whisper.py

```diff
@@ -0,0 +1,866 @@
+    all_model_classes = (TFWhisperModel, TFWhisperForConditionalGeneration) if is_tf_available() else ()
+    all_generative_model_classes = (TFWhisperForConditionalGeneration,) if is_tf_available() else ()
+    is_encoder_decoder = True
+    fx_compatible = False
+    test_pruning = False
+    test_missing_keys = False
+    test_onnx = False
+
+    input_name = "input_features"
+
+    def setUp(self):
+        self.model_tester = TFWhisperModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=WhisperConfig)
+        self.maxDiff = 3000
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_save_load_strict(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs()
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+
+            model(model.dummy_inputs)
+
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                model.save_pretrained(tmpdirname, saved_model=False)
+                model2, info = model_class.from_pretrained(tmpdirname, output_loading_info=True)
+            self.assertEqual(info["missing_keys"], [])
+
+    def test_model_forward(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model_forward(*config_and_inputs)
+
+    def test_decoder_model_past_with_large_inputs(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_decoder_model_past_large_inputs(*config_and_inputs)
+
+    def _get_input_ids_and_config(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        input_ids = inputs_dict[self.input_name]
+
+        # cut to half length & take max batch_size 3
+        max_batch_size = 3
+        input_ids = input_ids[:max_batch_size, :, :]
+
+        # generate max 3 tokens
+        max_length = input_ids.shape[-1] + 3
+        if config.eos_token_id is not None and config.pad_token_id is None:
+            # hack to allow generate for models such as GPT2 as is done in `generate()`
+            config.pad_token_id = config.eos_token_id
+
+        return config, input_ids, None, max_length
+
+    # not implemented currently
+    def test_inputs_embeds(self):
+        pass
+
+    unittest.skip("Training is not yet supported")
+
+    def test_training(self):
+        pass
+
+    def test_generate_with_head_masking(self):
+        pass
+
+    def test_generate_fp16(self):
+        config, input_dict = self.model_tester.prepare_config_and_inputs()
+        config.max_target_positions = 400
+        input_features = input_dict["input_features"]
+        model = TFWhisperForConditionalGeneration(config)
+        model.generate(input_features)
+        model.generate(input_features, num_beams=4, do_sample=True, early_stopping=False, num_return_sequences=3)
```

Comment:
I would skip this test for now, HF+TF+16bit support is very low atm

Reply:
But would have been good on PT side